relay-server/src/actors/processor.rs

```diff
@@ -2156,8 +2197,12 @@ impl EnvelopeProcessorService {
                         // requires recomputation of the context.
                         state.envelope_context.update(&state.envelope);
 
+                        let has_metrics = state.extracted_metrics.project_metrics.is_empty();
+
+                        state.extracted_metrics.send_metrics(&state.envelope);
+
                         let envelope_response = if state.envelope.is_empty() {
-                            if state.extracted_metrics.is_empty() {
+                            if !has_metrics {
```

Comment:
@jjbayer what about this line? 
if `has_metrics = !state.extracted_metrics.project_metrics.is_empty(); `
do we want to change it here?